Add rate limiting guides (#254)
agentgateway/website: assets/agw-docs/pages/security/rate-limit-global.md

4. The Rate Limit Service checks its configuration for matching descriptors, applies the configured limits, and returns a decision.
5. If allowed, the Gateway forwards the request to the App.
6. If the rate limit is exceeded, the Gateway denies the request with a 429 response and rate limit headers.

### Architecture

Global rate limiting requires two components:

1. **{{< reuse "agw-docs/snippets/trafficpolicy.md" >}} with `rateLimit.global`**: Configure your rate limit policy with descriptors that extract request attributes using CEL expressions. The policy specifies:
   - `backendRef`: Reference to the rate limit service (Service or Backend)
   - `domain`: Arbitrary string identifying this application/team
   - `descriptors`: Array of CEL-based rules for extracting rate limit dimensions

2. **Rate Limit Service**: An external service implementing the Envoy Rate Limit gRPC protocol. The service:
   - Stores the actual rate limit values (requests per time unit)
   - Maintains counters in a backend store (typically Redis)
   - Returns allow/deny decisions based on descriptor matching

> Unlike kgateway, agentgateway does **not** require a separate GatewayExtension resource — the policy directly references the rate limit service.

Reviewer: I think we can maybe cut this architecture section or incorporate pieces further up

| User agent | `request.headers["user-agent"]` | Client user agent string. |

## Before you begin

{{< reuse "agw-docs/snippets/prereq-x-channel.md" >}}

## Deploy the rate limit service {#deploy-service}

You need an external rate limit service that implements the Envoy Rate Limit gRPC protocol. This example uses the reference implementation from Envoy with Redis as the backend store.

1. Create a namespace for the rate limit infrastructure.

   ```sh
   kubectl create namespace ratelimit
   ```

2. Deploy Redis as the backing store.

   ```yaml
   kubectl apply -f- <<EOF
   apiVersion: apps/v1
   kind: Deployment
   metadata:
     name: redis
     namespace: ratelimit
   spec:
     replicas: 1
     selector:
       matchLabels:
         app: redis
     template:
       metadata:
         labels:
           app: redis
       spec:
         containers:
           - name: redis
             image: redis:7-alpine
             ports:
               - containerPort: 6379
   ---
   apiVersion: v1
   kind: Service
   metadata:
     name: redis
     namespace: ratelimit
   spec:
     selector:
       app: redis
     ports:
       - port: 6379
   EOF
   ```

> 3. Create a ConfigMap with rate limit rules. This configuration defines the actual rate limits that will be enforced.

Reviewer: ```suggestion
3. Create a ConfigMap with rate limit rules. This configuration defines the actual rate limits that are enforced by the rate limit service.
```

     wait:
       target:
         apiVersion: agentgateway.dev/v1alpha1
         kind: AgentgatewayPolicy
         metadata:
           namespace: httpbin
           name: ip-rate-limit
       jsonPath: "$.status.ancestors[0].conditions[?(@.type=='Accepted')].status"
       jsonPathExpectation:
         comparator: equals
         value: "True"
       polling:
         timeoutSeconds: 120
         intervalSeconds: 2
   EOF
   {{< /doc-test >}}

2. Send 15 rapid requests from the same client.

   {{< tabs tabTotal="2" items="Cloud Provider LoadBalancer,Port-forward for local testing" >}}
   {{% tab tabName="Cloud Provider LoadBalancer" %}}
   ```sh
   for i in $(seq 1 15); do
     STATUS=$(curl -s -o /dev/null -w "%{http_code}" \
       http://$INGRESS_GW_ADDRESS:80/get \
       -H "host: www.example.com")
     echo "Request $i: HTTP $STATUS"
   done
   ```
   {{% /tab %}}
   {{% tab tabName="Port-forward for local testing" %}}
   ```sh
   for i in $(seq 1 15); do
     STATUS=$(curl -s -o /dev/null -w "%{http_code}" \
       localhost:8080/get \
       -H "host: www.example.com")
     echo "Request $i: HTTP $STATUS"
   done
   ```
   {{% /tab %}}
   {{< /tabs >}}

>    The first 10 requests succeed with 200, and subsequent requests return 429. All requests from the same IP share the same counter.

Reviewer: example output could be added here
